### ISSUE TITLE
Receiving video stream when video is not requested

### ISSUE THREAD

**Issue by lnogueir** [ant-media/Ant-Media-Server#3411]
### Receiving video stream when video is not requested

I have a stream that's been published with the following command:
```
{
    command : "publish",
    streamId : "someStreamId",
    token : "someTokenId",
    video: "true",
    audio: "true"
}
```

All is good and I am able to play this stream back. However, I have certain sessions that only require the audio stream. Due to this, I have been trying to **not** receive the video stream by sending the following command:
```
{
    command : "play",
    streamId : "someStreamId",
    token : "someTokenId",
    video: "false",
    audio: "true"
}
```

By sending the above command with the `video` key set to `false`, I am expecting to only receive an audio stream,  but I am still receiving the published video track on my list of tracks.

The Ant Media Server version I am using is: `v2.1.0`.

Is this a bug in the version I am using? If not, how can I get the server to only send me  **only** the **audio** stream?

**Comment by cemklc** [comment]
Hi,

I will be responsible for this issue from now on. I see that you have connected with the team on google groups as well, but we can continue from here.

Can you tell me what API are you using since I believe there are some differences with your commands.

What I understand is that you have a video stream with both video and audio, and you want to access the only audio part for some instances right?

I will update once I am sure that I have found a stable way.

**Comment by lnogueir** [comment]
> Hi,
> 
> I will be responsible for this issue from now on. I see that you have connected with the team on google groups as well, but we can continue from here.
> 
> Can you tell me what API are you using since I believe there are some differences with your commands.
> 
> What I understand is that you have a video stream with both video and audio, and you want to access the only audio part for some instances right?
> 
> I will update once I am sure that I have found a stable way.

Hi,

I'm using the WebRTC signalling WebSocket API version `v2.1.0`.

And yes, that is correct. I have streams published with audio and video, but in some instances, I only need to receive audio. But right now, audio and video are coming. My current workaround is to have a fake decoder that just ignores the incoming video frames, but this isn't ideal at all. What I would like to do is establish a session with only audio, and not receive any video frames from the network.

Let me know if you have any updates.

Thanks!

**Comment by cemklc** [comment]
Hello, 

There is a built-in command in Ant Media for toggling both audio and video. Here is the repository;
https://github.com/ant-media/StreamApp/blob/master/src/main/webapp/js/webrtc_adaptor.js

You can use this command for disabling the video; StreamId and the tracking id should be the same if you aren't using multi-track. For the enabled command you can input true or false depending on your needs.

{
    command : "toggleVideo",
    streamId: streamId,
    trackId: trackId,
    enabled: enabled,
}

Regards,
Cem

**Comment by lnogueir** [comment]
I had taken a look at this `toggle`, but it doesn't seem like this code is available on version `v2.1.0`, which is the one I'm using and can't upgrade right now.

Is there any other way to achieve that on the version I'm using?

**Comment by cemklc** [comment]
Hello,

Unfortunately, there are only workaround solutions like you did, there isn't any way that is developed by Antmedia. 

Newer versions came with a lot of optimizations and new features, we recommend updating the server whenever possible.
